client/src/components/ResultPanel/ResultPanel.ts

```diff
@@ -4,63 +4,116 @@ import { Uri, ViewColumn, WebviewPanel, l10n, window } from "vscode";
 
-let resultPanel: WebviewPanel | undefined;
-export const resultPanels: Record<string, WebviewPanel> = {};
+const SCRIPT_REGEX = /<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi;
+export const SAS_RESULT_PANEL = "SASResultPanel";
+
+interface ResultPanelState {
+  panelId: string;
+}
+
+interface IdentifiableWebviewPanel {
+  webviewPanel: WebviewPanel;
+  panelId: string;
+}
+
+let resultPanel: IdentifiableWebviewPanel | undefined;
 
 export const showResult = (html: string, uri?: Uri, title?: string) => {
-  html = html
-    // Inject vscode context into our results html body
-    .replace(
-      "<body ",
-      `<body data-vscode-context='${JSON.stringify({
-        preventDefaultContextMenuItems: true,
-      })}' `,
-    )
-    // Make sure the html and body take up the full height of the parent
-    // iframe so that the context menu is clickable anywhere on the page
-    .replace(
-      "</head>",
-      "<style>html,body { height: 100% !important; }</style></head>",
-    );
   const sideResult = isSideResultEnabled();
   const singlePanel = isSinglePanelEnabled();
+  let panelId: string;
+
   if (!title) {
     title = l10n.t("Result");
   }
 
   if (!singlePanel || !resultPanel) {
-    const resultPanelId = `SASResultPanel-${v4()}`;
-    resultPanel = window.createWebviewPanel(
-      resultPanelId, // Identifies the type of the webview. Used internally
+    panelId = `${v4()}`;
+    const webviewPanel = window.createWebviewPanel(
+      SAS_RESULT_PANEL, // Identifies the type of the webview. Used internally
       title, // Title of the panel displayed to the user
       {
         preserveFocus: true,
         viewColumn: sideResult ? ViewColumn.Beside : ViewColumn.Active,
       }, // Editor column to show the new webview panel in.
-      {}, // Webview options. More on these later.
-    );
-    resultPanel.onDidDispose(
-      ((id) => () => {
-        delete resultPanels[id];
-        resultPanel = undefined;
-      })(resultPanelId),
+      { enableScripts: true }, // Webview options.
     );
-    resultPanels[resultPanelId] = resultPanel;
+    webviewPanel.onDidDispose(() => disposePanel(panelId));
+    resultPanel = { webviewPanel, panelId };
   } else {
     const editor = uri
       ? window.visibleTextEditors.find(
           (editor) => editor.document.uri.toString() === uri.toString(),
         )
       : window.activeTextEditor;
-    if (resultPanel.title !== title) {
-      resultPanel.title = title;
+    if (resultPanel.webviewPanel.title !== title) {
+      resultPanel.webviewPanel.title = title;
     }
-    resultPanel.reveal(
+    panelId = resultPanel.panelId;
+    resultPanel.webviewPanel.reveal(
       sideResult ? ViewColumn.Beside : editor?.viewColumn,
       true,
     );
   }
-  resultPanel.webview.html = html;
+
+  const panelHtml = wrapPanelHtml(html, panelId);
+  resultPanel.webviewPanel.webview.html = panelHtml;
+  setContextValue(resultPanel.panelId, panelHtml);
+};
+
+/**
```

Comment:
Would you either complete the comment or remove the comment?

Reply:
Removed in https://github.com/sassoftware/vscode-sas-extension/pull/935/commits/1708088bbdaee17dfac035c7896e2b7f1fdaf35a